src/main/java/de/adesso/blogpostchecker/CheckExecutor.java

```diff
@@ -9,89 +9,82 @@
-        checkTagsNotEmpty(postHeader);
-        checkAuthorNotEmpty(postHeader);
-        checkAuthorInYml(postHeader, authors);
-        checkTitleNotEmpty(postHeader);
-        checkLayoutCorrect(postHeader);
-        checkDateMatchesFormat(postHeader);
+    public void executeChecks(PostMetaData postMetaData, String authors) {
+        checkCategoryNotEmpty(postMetaData);
+        checkCategoryOnlyOneEntry(postMetaData);
+        checkTagsNotEmpty(postMetaData);
+        checkAuthorNotEmpty(postMetaData);
+        checkAuthorInYml(postMetaData, authors);
+        checkTitleNotEmpty(postMetaData);
+        checkLayoutCorrect(postMetaData);
+        checkDateMatchesFormat(postMetaData);
     }
 
-    private void checkCategoryOnlyOneEntry(PostHeader header) {
-        if (header.getCategories() != null && !header.getCategories().equals("")) {
-            if (!header.getCategories().contains(",")) {
-                LOGGER.info("Categories checked");
-            } else {
-                LOGGER.error("There are two or more categories.");
-                LOGGER.error("Exiting BlogpostChecker.");
-                System.exit(31);
-            }
+    private void checkCategoryNotEmpty(PostMetaData metaData) {
+        if (checkAttribute(metaData.getCategories())) {
+            LOGGER.info("Categories not empty checked.");
         } else {
-            LOGGER.error("Categories does not match.");
-            LOGGER.error("Exiting BlogpostChecker.");
-            System.exit(30);
+            ExitBlogpostChecker.exit(LOGGER, "THe categories does not match.", 30);
         }
     }
 
-    private void checkTagsNotEmpty(PostHeader header) {
-        if (header.getTags() != null && !header.getTags().equals("")) {
+    private void checkCategoryOnlyOneEntry(PostMetaData metaData) {
+        if (!metaData.getCategories().contains(",")) {
+            LOGGER.info("Categories checked");
+        } else {
+            ExitBlogpostChecker.exit(LOGGER, "There are two or more categories.", 31);
+        }
+    }
+
+    private void checkTagsNotEmpty(PostMetaData metaData) {
+        if (checkAttribute(metaData.getTags())) {
             LOGGER.info("Tags checked");
         } else {
-            LOGGER.error("Tags are empty");
-            LOGGER.error("Exiting BlogpostChecker.");
-            System.exit(32);
+            ExitBlogpostChecker.exit(LOGGER, "The tags are empty.", 32);
         }
     }
 
-    private void checkAuthorNotEmpty(PostHeader header) {
-        if (header.getAuthor() != null && !header.getAuthor().equals("")) {
+    private void checkAuthorNotEmpty(PostMetaData metaData) {
+        if (checkAttribute(metaData.getAuthor())) {
             LOGGER.info("Author checked");
         } else {
-            LOGGER.error("Author is empty.");
-            LOGGER.error("Exiting BlogpostChecker.");
-            System.exit(33);
+            ExitBlogpostChecker.exit(LOGGER, "The author ist empty.", 33);
```

Comment:
```suggestion
            ExitBlogpostChecker.exit(LOGGER, "No author found. Exactly one author expected.", 33);
```
Dürfen Artikel mehr als einen Author besitzen?